PubNub/Data/PNConfiguration.m

```diff
@@ -193,13 +193,13 @@ - (instancetype)initWithPublishKey:(NSString *)publishKey
 
 - (id)copyWithZone:(NSZone *)zone {
     PNConfiguration *configuration = [[PNConfiguration allocWithZone:zone] init];
-    configuration.deviceID = self.deviceID;
-    configuration.origin = self.origin;
-    configuration.publishKey = self.publishKey;
-    configuration.subscribeKey = self.subscribeKey;
-    configuration.authKey = self.authKey;
-    configuration.authToken = self.authToken;
-    configuration.userID = self.userID;
+    configuration.deviceID = [self.deviceID copy];
+    configuration.origin = [self.origin copy];
+    configuration.publishKey = [self.publishKey copy];
+    configuration.subscribeKey = [self.subscribeKey copy];
+    configuration.authKey = [self.authKey copy];
+    configuration.authToken = [self.authToken copy];
+    configuration.userID = [self.userID copy];
     configuration.cryptoModule = self.cryptoModule;
```

Comment:
I know it's not related with requests retrying, but shouldn't you also make a deep copy of `cryptoModule`?

Reply:
@jguz-pubnub there is no need in copy because it is stateless and there is no way for user to change anything after it has been passed into `PNConfiguration`.